File: projects/m1/018-roulette-payouts/solutions/stefandifede/roulette-payouts.js
//creo un array con tutti i valori della roulette
const spaces = ["00", ...Array(37).keys()];

//genero un indice dell'array per trovare il numero vincente casuale 
const randomIndex = Math.floor(Math.random()*spaces.length);
const winningValue = spaces[randomIndex];    

//payouts: single number, red vs black, odd or even, 1-18 vs 19-36
//red vs black
const redSpaces = [1, 3, 5, 7, 9, 12, 14, 16, 18, 19, 21, 23, 25, 27, 30, 32, 34, 36]
let winningColor = " ";
function redVsBlack () {
    if (redSpaces.includes(winningValue)) {
        winningColor = "Red";
    } else (winningColor = "Black")
}

redVsBlack();


//odd or even
let winningOddOrEven = " ";
function oddOrEven () {
    if (winningValue % 2 == 0) {
        winningOddOrEven = "Even"
    } else (winningOddOrEven = "Odd")
}

oddOrEven();

//1-18 vs 19-36 
let winningLowVsHigh = " ";
function lowVsHigh() {
    if (winningValue < 19) {
        winningLowVsHigh = "1 to 18"
    } else (winningLowVsHigh = "19 to 36")
}

lowVsHigh()


//divido le possibilità di vittoria tra 0,00 e tutte le restanti possibilità, in quanto 0 e 00 devono pagare solotanto le scommesse sul numero esatto
function declaringPayouts() {
    if (winningValue == 0 || winningValue == 00) {
        alert("The spin resulted in: " + winningValue + "\nPay " + winningValue)
    } else (alert ("The spin resulted in: " + winningValue + "\nPay " + winningValue + "\nPay " + winningColor + "\nPay " + winningOddOrEven + "\nPay " + winningLowVsHigh))
}

declaringPayouts(); 

File: projects/m1/018-roulette-payouts/solutions/Davide Di Giacomo/roulette_payouts.py
import random

#variables
number_list = [0, 28, 9, 26, 30, 11, 7, 20, 32, 17, 5, 22, 34, 15, 3, 24, 36, 13, 1, 00, 27, 10, 25, 29, 12, 8, 19, 31, 18, 6, 21, 33, 16, 4, 23, 35, 14, 2]
red = [1, 3, 5, 7, 9, 12, 14, 16, 18, 19, 21, 23, 25, 27, 30, 32, 34, 36]
black = [2, 4, 6, 8, 10, 11, 13, 15, 17, 20, 22, 24, 26, 28, 29, 31, 33, 35]

#random number generator
number = random.choice(number_list)
print (f"The spin resulted in {number}...")
print (f"pay {number}")

#Red/Black 
if number in red:
    print ("pay Red")   
elif number in black:
    print ("pay Black")
    
#Odd/Even
if number % 2 == 0 and number != 0 and number != 00:
    print ("pay Even")  
elif number % 2 != 0 and number != 0 and number != 00:
    print ("pay Odd")
    
#1 to 18/19 to 36
if number < 19 and number != 0 and number != 00:
    print ("pay 1 to 18")
elif number > 18:
    print ("pay 19 to 36") 